craft_cli/dispatcher.py

```diff
@@ -93,6 +116,9 @@ class GlobalArgument(NamedTuple):
         None,
         "--verbosity",
         "Set the verbosity level to 'quiet', 'brief', 'verbose', 'debug' or 'trace'",
+        choices=[mode.name.lower() for mode in EmitterMode],
+        validator=lambda mode: EmitterMode[mode.upper()],
+        case_sensitive=False,
```

Comment:
`choices` also exists in argparse, will `validator` and `case_sensitive` options increase the gap between global argument handling and command-specific argument handling instead of reducing it? Ideally we should have an uniform way to handle all arguments.

Reply:
`validator` is equivalent to argparse's `type` - I can rename if you'd like.

`case_sensitive` is different, but is there for backwards compatibility in the way we handle `--verbosity`, which was previously case-insensitive due to special handling. Looking over it again, I wonder if I could just run the validator before limiting choices and make a special `verbosity_validator` function that returns the correct emitter mode.